Make pylint pass

<!-- ⚠️ If you do not respect this template, your issue will be closed -->
<!-- ⚠️ Make sure to browse the opened and closed issues to confirm this idea does not exist. -->

### What is the expected enhancement?

To merge code into Qiskit, we have to make pylint pass without an error. You can check pylint by running
```
make lint   # global for all of Qiskit Terra, takes pretty long (~20min for me)
pylint -rn qiskit/transpiler/passes/synthesis test/python/transpiler/test_solovay_kitaev.py  # our files only, very fast
```
alternatively, check it with VS Code 🙂 

<details closed>
<summary>The current output is:</summary>
<br>

```
pylint -rn qiskit/transpiler/passes/synthesis test/python/transpiler/test_solovay_kitaev.py
************* Module qiskit.transpiler.passes.synthesis
qiskit/transpiler/passes/synthesis/__init__.py:19:0: C0304: Final newline missing (missing-final-newline)
************* Module qiskit.transpiler.passes.synthesis.solovay_kitaev_utils
qiskit/transpiler/passes/synthesis/solovay_kitaev_utils.py:390:0: C0103: Argument name "u" doesn't conform to '[a-z_][a-z0-9_]{2,30}|ax|dt$' pattern (invalid-name)
qiskit/transpiler/passes/synthesis/solovay_kitaev_utils.py:402:0: C0103: Argument name "u" doesn't conform to '[a-z_][a-z0-9_]{2,30}|ax|dt$' pattern (invalid-name)
************* Module qiskit.transpiler.passes.synthesis.solovay_kitaev
qiskit/transpiler/passes/synthesis/solovay_kitaev.py:58:0: C0330: Wrong continued indentation (add 1 space).
                            for comb in itertools.product(*[basis_gates] * reps))
                            ^| (bad-continuation)
************* Module test.python.transpiler.test_solovay_kitaev
test/python/transpiler/test_solovay_kitaev.py:35:59: C0303: Trailing whitespace (trailing-whitespace)
test/python/transpiler/test_solovay_kitaev.py:150:30: C0326: Exactly one space required after :
def _generate_x_rotation(angle:float) -> np.ndarray:
                              ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:151:0: C0301: Line too long (103/100) (line-too-long)
test/python/transpiler/test_solovay_kitaev.py:151:23: C0326: Exactly one space required after comma
    return np.array([[1,0,0],[0,math.cos(angle),-math.sin(angle)],[0,math.sin(angle),math.cos(angle)]])
                       ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:151:25: C0326: Exactly one space required after comma
    return np.array([[1,0,0],[0,math.cos(angle),-math.sin(angle)],[0,math.sin(angle),math.cos(angle)]])
                         ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:151:28: C0326: Exactly one space required after comma
    return np.array([[1,0,0],[0,math.cos(angle),-math.sin(angle)],[0,math.sin(angle),math.cos(angle)]])
                            ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:151:31: C0326: Exactly one space required after comma
    return np.array([[1,0,0],[0,math.cos(angle),-math.sin(angle)],[0,math.sin(angle),math.cos(angle)]])
                               ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:151:47: C0326: Exactly one space required after comma
    return np.array([[1,0,0],[0,math.cos(angle),-math.sin(angle)],[0,math.sin(angle),math.cos(angle)]])
                                               ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:151:65: C0326: Exactly one space required after comma
    return np.array([[1,0,0],[0,math.cos(angle),-math.sin(angle)],[0,math.sin(angle),math.cos(angle)]])
                                                                 ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:151:68: C0326: Exactly one space required after comma
    return np.array([[1,0,0],[0,math.cos(angle),-math.sin(angle)],[0,math.sin(angle),math.cos(angle)]])
                                                                    ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:151:84: C0326: Exactly one space required after comma
    return np.array([[1,0,0],[0,math.cos(angle),-math.sin(angle)],[0,math.sin(angle),math.cos(angle)]])
                                                                                    ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:153:30: C0326: Exactly one space required after :
def _generate_y_rotation(angle:float) -> np.ndarray:
                              ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:154:0: C0301: Line too long (103/100) (line-too-long)
test/python/transpiler/test_solovay_kitaev.py:154:37: C0326: Exactly one space required after comma
    return np.array([[math.cos(angle),0,math.sin(angle)],[0,1,0],[-math.sin(angle),0,math.cos(angle)]])
                                     ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:154:39: C0326: Exactly one space required after comma
    return np.array([[math.cos(angle),0,math.sin(angle)],[0,1,0],[-math.sin(angle),0,math.cos(angle)]])
                                       ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:154:56: C0326: Exactly one space required after comma
    return np.array([[math.cos(angle),0,math.sin(angle)],[0,1,0],[-math.sin(angle),0,math.cos(angle)]])
                                                        ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:154:59: C0326: Exactly one space required after comma
    return np.array([[math.cos(angle),0,math.sin(angle)],[0,1,0],[-math.sin(angle),0,math.cos(angle)]])
                                                           ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:154:61: C0326: Exactly one space required after comma
    return np.array([[math.cos(angle),0,math.sin(angle)],[0,1,0],[-math.sin(angle),0,math.cos(angle)]])
                                                             ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:154:64: C0326: Exactly one space required after comma
    return np.array([[math.cos(angle),0,math.sin(angle)],[0,1,0],[-math.sin(angle),0,math.cos(angle)]])
                                                                ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:154:82: C0326: Exactly one space required after comma
    return np.array([[math.cos(angle),0,math.sin(angle)],[0,1,0],[-math.sin(angle),0,math.cos(angle)]])
                                                                                  ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:154:84: C0326: Exactly one space required after comma
    return np.array([[math.cos(angle),0,math.sin(angle)],[0,1,0],[-math.sin(angle),0,math.cos(angle)]])
                                                                                    ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:156:30: C0326: Exactly one space required after :
def _generate_z_rotation(angle:float) -> np.ndarray:
                              ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:157:0: C0301: Line too long (103/100) (line-too-long)
test/python/transpiler/test_solovay_kitaev.py:157:37: C0326: Exactly one space required after comma
    return np.array([[math.cos(angle),-math.sin(angle),0],[math.sin(angle),math.cos(angle),0],[0,0,1]])
                                     ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:157:54: C0326: Exactly one space required after comma
    return np.array([[math.cos(angle),-math.sin(angle),0],[math.sin(angle),math.cos(angle),0],[0,0,1]])
                                                      ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:157:57: C0326: Exactly one space required after comma
    return np.array([[math.cos(angle),-math.sin(angle),0],[math.sin(angle),math.cos(angle),0],[0,0,1]])
                                                         ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:157:74: C0326: Exactly one space required after comma
    return np.array([[math.cos(angle),-math.sin(angle),0],[math.sin(angle),math.cos(angle),0],[0,0,1]])
                                                                          ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:157:90: C0326: Exactly one space required after comma
    return np.array([[math.cos(angle),-math.sin(angle),0],[math.sin(angle),math.cos(angle),0],[0,0,1]])
                                                                                          ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:157:93: C0326: Exactly one space required after comma
    return np.array([[math.cos(angle),-math.sin(angle),0],[math.sin(angle),math.cos(angle),0],[0,0,1]])
                                                                                             ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:157:96: C0326: Exactly one space required after comma
    return np.array([[math.cos(angle),-math.sin(angle),0],[math.sin(angle),math.cos(angle),0],[0,0,1]])
                                                                                                ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:157:98: C0326: Exactly one space required after comma
    return np.array([[math.cos(angle),-math.sin(angle),0],[math.sin(angle),math.cos(angle),0],[0,0,1]])
                                                                                                  ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:174:28: C0326: Exactly one space required after comma
        return np.array([1.0,0.0,0.0])
                            ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:174:32: C0326: Exactly one space required after comma
        return np.array([1.0,0.0,0.0])
                                ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:176:28: C0326: Exactly one space required after comma
        return np.array([0.0,1.0,0.0])
                            ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:176:32: C0326: Exactly one space required after comma
        return np.array([0.0,1.0,0.0])
                                ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:178:28: C0326: Exactly one space required after comma
        return np.array([0.0,0.0,1.0])
                            ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:178:32: C0326: Exactly one space required after comma
        return np.array([0.0,0.0,1.0])
                                ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:180:39: C0326: Exactly one space required after comma
        return np.array([1/math.sqrt(3),1/math.sqrt(3),1/math.sqrt(3)])
                                       ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:180:54: C0326: Exactly one space required after comma
        return np.array([1/math.sqrt(3),1/math.sqrt(3),1/math.sqrt(3)])
                                                      ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:182:25: C0326: Exactly one space required after :
def _generate_x_su2(angle:float) -> np.ndarray:
                         ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:184:0: C0330: Wrong continued indentation (remove 2 spaces).
                       [math.sin(angle/2)*1j, math.cos(angle/2)]], dtype=complex)
                     | ^ (bad-continuation)
test/python/transpiler/test_solovay_kitaev.py:186:25: C0326: Exactly one space required after :
def _generate_y_su2(angle:float) -> np.ndarray:
                         ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:188:0: C0330: Wrong continued indentation (remove 4 spaces).
                         [-math.sin(angle/2), math.cos(angle/2)]], dtype=complex)
                     |   ^ (bad-continuation)
test/python/transpiler/test_solovay_kitaev.py:190:25: C0326: Exactly one space required after :
def _generate_z_su2(angle:float) -> np.ndarray:
                         ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:194:27: C0326: Exactly one space required after comma
    base = np.array([[alpha,beta],[-np.conj(beta),np.conj(alpha)]])
                           ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:194:33: C0326: Exactly one space required after comma
    base = np.array([[alpha,beta],[-np.conj(beta),np.conj(alpha)]])
                                 ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:194:49: C0326: Exactly one space required after comma
    base = np.array([[alpha,beta],[-np.conj(beta),np.conj(alpha)]])
                                                 ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:196:15: C0326: Exactly one space required around comparison
    if abs(det)<1e10:
               ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:197:27: C0326: Exactly one space required after comma
        return np.array([[1,0],[0,1]])
                           ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:197:30: C0326: Exactly one space required after comma
        return np.array([[1,0],[0,1]])
                              ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:197:33: C0326: Exactly one space required after comma
        return np.array([[1,0],[0,1]])
                                 ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:202:24: C0326: Exactly one space required after comma
    vector = np.array([a,b,c])
                        ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:202:26: C0326: Exactly one space required after comma
    vector = np.array([a,b,c])
                          ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:203:32: C0326: Exactly one space required before comparison
    if a != 0.0 or b != 0.0 or c!= 0.0:
                                ^^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:207:26: C0326: Exactly one space required after comma
        return np.array([1,0,0])
                          ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:207:28: C0326: Exactly one space required after comma
        return np.array([1,0,0])
                            ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:210:106: C0303: Trailing whitespace (trailing-whitespace)
test/python/transpiler/test_solovay_kitaev.py:210:0: C0301: Line too long (106/100) (line-too-long)
test/python/transpiler/test_solovay_kitaev.py:210:28: C0326: Exactly one space required after comma
    return array.shape == (3,3) and abs(np.linalg.det(array)-1.0)< 1e-10 and not False in np.isreal(array)
                            ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:210:65: C0326: Exactly one space required before comparison
    return array.shape == (3,3) and abs(np.linalg.det(array)-1.0)< 1e-10 and not False in np.isreal(array)
                                                                 ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:213:39: C0326: Exactly one space required after comma
    for t in itertools.product(range(2),range(2)):
                                       ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:214:43: C0326: Exactly one space required before comparison
        if abs(a[t[0]][t[1]]-b[t[0]][t[1]])> 1e-10:
                                           ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:226:41: C0326: Exactly one space required after comma
        [np.dot(_generate_z_rotation(0.5),_generate_y_rotation(0.4))],
                                         ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:227:41: C0326: Exactly one space required after comma
        [np.dot(_generate_y_rotation(0.5),_generate_x_rotation(0.4))]
                                         ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:230:98: C0303: Trailing whitespace (trailing-whitespace)
test/python/transpiler/test_solovay_kitaev.py:235:0: C0301: Line too long (118/100) (line-too-long)
test/python/transpiler/test_solovay_kitaev.py:235:36: C0326: Exactly one space required after comma
    @given(st.builds(_build_rotation,st.floats(max_value=2*math.pi,min_value=0),st.integers(min_value=0,max_value=4)))
                                    ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:235:66: C0326: Exactly one space required after comma
    @given(st.builds(_build_rotation,st.floats(max_value=2*math.pi,min_value=0),st.integers(min_value=0,max_value=4)))
                                                                  ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:235:79: C0326: Exactly one space required after comma
    @given(st.builds(_build_rotation,st.floats(max_value=2*math.pi,min_value=0),st.integers(min_value=0,max_value=4)))
                                                                               ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:235:103: C0326: Exactly one space required after comma
    @given(st.builds(_build_rotation,st.floats(max_value=2*math.pi,min_value=0),st.integers(min_value=0,max_value=4)))
                                                                                                       ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:236:100: C0303: Trailing whitespace (trailing-whitespace)
test/python/transpiler/test_solovay_kitaev.py:245:41: C0326: Exactly one space required after comma
        [np.dot(_generate_z_rotation(0.5),_generate_y_rotation(0.4))],
                                         ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:246:41: C0326: Exactly one space required after comma
        [np.dot(_generate_y_rotation(0.5),_generate_x_rotation(0.4))]
                                         ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:249:103: C0303: Trailing whitespace (trailing-whitespace)
test/python/transpiler/test_solovay_kitaev.py:249:0: C0301: Line too long (103/100) (line-too-long)
test/python/transpiler/test_solovay_kitaev.py:253:0: C0301: Line too long (125/100) (line-too-long)
test/python/transpiler/test_solovay_kitaev.py:253:44: C0326: Exactly one space required after comma
        actual_commutator = np.dot(first_so3,np.dot(second_so3,np.dot(np.matrix.getH(first_so3),np.matrix.getH(second_so3))))
                                            ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:253:62: C0326: Exactly one space required after comma
        actual_commutator = np.dot(first_so3,np.dot(second_so3,np.dot(np.matrix.getH(first_so3),np.matrix.getH(second_so3))))
                                                              ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:253:95: C0326: Exactly one space required after comma
        actual_commutator = np.dot(first_so3,np.dot(second_so3,np.dot(np.matrix.getH(first_so3),np.matrix.getH(second_so3))))
                                                                                               ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:254:53: C0326: No space allowed after bracket
        self.assertTrue(are_almost_equal_so3_matrices( actual_commutator,u_so3))
                                                     ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:254:72: C0326: Exactly one space required after comma
        self.assertTrue(are_almost_equal_so3_matrices( actual_commutator,u_so3))
                                                                        ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:256:0: C0301: Line too long (118/100) (line-too-long)
test/python/transpiler/test_solovay_kitaev.py:256:36: C0326: Exactly one space required after comma
    @given(st.builds(_build_rotation,st.floats(max_value=2*math.pi,min_value=0),st.integers(min_value=0,max_value=4)))
                                    ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:256:66: C0326: Exactly one space required after comma
    @given(st.builds(_build_rotation,st.floats(max_value=2*math.pi,min_value=0),st.integers(min_value=0,max_value=4)))
                                                                  ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:256:79: C0326: Exactly one space required after comma
    @given(st.builds(_build_rotation,st.floats(max_value=2*math.pi,min_value=0),st.integers(min_value=0,max_value=4)))
                                                                               ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:256:103: C0326: Exactly one space required after comma
    @given(st.builds(_build_rotation,st.floats(max_value=2*math.pi,min_value=0),st.integers(min_value=0,max_value=4)))
                                                                                                       ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:257:105: C0303: Trailing whitespace (trailing-whitespace)
test/python/transpiler/test_solovay_kitaev.py:257:0: C0301: Line too long (105/100) (line-too-long)
test/python/transpiler/test_solovay_kitaev.py:261:0: C0301: Line too long (125/100) (line-too-long)
test/python/transpiler/test_solovay_kitaev.py:261:44: C0326: Exactly one space required after comma
        actual_commutator = np.dot(first_so3,np.dot(second_so3,np.dot(np.matrix.getH(first_so3),np.matrix.getH(second_so3))))
                                            ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:261:62: C0326: Exactly one space required after comma
        actual_commutator = np.dot(first_so3,np.dot(second_so3,np.dot(np.matrix.getH(first_so3),np.matrix.getH(second_so3))))
                                                              ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:261:95: C0326: Exactly one space required after comma
        actual_commutator = np.dot(first_so3,np.dot(second_so3,np.dot(np.matrix.getH(first_so3),np.matrix.getH(second_so3))))
                                                                                               ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:262:53: C0326: No space allowed after bracket
        self.assertTrue(are_almost_equal_so3_matrices( actual_commutator,u_so3))
                                                     ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:262:72: C0326: Exactly one space required after comma
        self.assertTrue(are_almost_equal_so3_matrices( actual_commutator,u_so3))
                                                                        ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:264:0: C0301: Line too long (118/100) (line-too-long)
test/python/transpiler/test_solovay_kitaev.py:264:36: C0326: Exactly one space required after comma
    @given(st.builds(_build_rotation,st.floats(max_value=2*math.pi,min_value=0),st.integers(min_value=0,max_value=4)))
                                    ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:264:66: C0326: Exactly one space required after comma
    @given(st.builds(_build_rotation,st.floats(max_value=2*math.pi,min_value=0),st.integers(min_value=0,max_value=4)))
                                                                  ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:264:79: C0326: Exactly one space required after comma
    @given(st.builds(_build_rotation,st.floats(max_value=2*math.pi,min_value=0),st.integers(min_value=0,max_value=4)))
                                                                               ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:264:103: C0326: Exactly one space required after comma
    @given(st.builds(_build_rotation,st.floats(max_value=2*math.pi,min_value=0),st.integers(min_value=0,max_value=4)))
                                                                                                       ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:268:43: C0326: Exactly one space required after comma
        self.assertAlmostEqual(actual[0][0],1.0)
                                           ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:269:43: C0326: Exactly one space required after comma
        self.assertAlmostEqual(actual[0][1],0.0)
                                           ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:270:43: C0326: Exactly one space required after comma
        self.assertAlmostEqual(actual[0][2],0.0)
                                           ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:271:43: C0326: Exactly one space required after comma
        self.assertAlmostEqual(actual[1][0],0.0)
                                           ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:272:43: C0326: Exactly one space required after comma
        self.assertAlmostEqual(actual[2][0],0.0)
                                           ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:274:0: C0301: Line too long (118/100) (line-too-long)
test/python/transpiler/test_solovay_kitaev.py:274:36: C0326: Exactly one space required after comma
    @given(st.builds(_build_rotation,st.floats(max_value=2*math.pi,min_value=0),st.integers(min_value=0,max_value=4)))
                                    ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:274:66: C0326: Exactly one space required after comma
    @given(st.builds(_build_rotation,st.floats(max_value=2*math.pi,min_value=0),st.integers(min_value=0,max_value=4)))
                                                                  ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:274:79: C0326: Exactly one space required after comma
    @given(st.builds(_build_rotation,st.floats(max_value=2*math.pi,min_value=0),st.integers(min_value=0,max_value=4)))
                                                                               ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:274:103: C0326: Exactly one space required after comma
    @given(st.builds(_build_rotation,st.floats(max_value=2*math.pi,min_value=0),st.integers(min_value=0,max_value=4)))
                                                                                                       ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:275:0: C0301: Line too long (101/100) (line-too-long)
test/python/transpiler/test_solovay_kitaev.py:278:43: C0326: Exactly one space required after comma
        self.assertAlmostEqual(actual[1][1],1.0)
                                           ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:279:43: C0326: Exactly one space required after comma
        self.assertAlmostEqual(actual[0][1],0.0)
                                           ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:280:43: C0326: Exactly one space required after comma
        self.assertAlmostEqual(actual[1][0],0.0)
                                           ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:281:43: C0326: Exactly one space required after comma
        self.assertAlmostEqual(actual[1][2],0.0)
                                           ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:282:43: C0326: Exactly one space required after comma
        self.assertAlmostEqual(actual[2][1],0.0)
                                           ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:283:0: C0303: Trailing whitespace (trailing-whitespace)
test/python/transpiler/test_solovay_kitaev.py:342:34: C0326: Exactly one space required after comma
    @data([GateSequence([IGate()]),IGate(),GateSequence([IGate(),IGate()])])
                                  ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:342:42: C0326: Exactly one space required after comma
    @data([GateSequence([IGate()]),IGate(),GateSequence([IGate(),IGate()])])
                                          ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:342:64: C0326: Exactly one space required after comma
    @data([GateSequence([IGate()]),IGate(),GateSequence([IGate(),IGate()])])
                                                                ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:344:24: C0326: Exactly one space required after comma
    def test_append(self,first_value,second_value,third_value):
                        ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:344:36: C0326: Exactly one space required after comma
    def test_append(self,first_value,second_value,third_value):
                                    ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:344:49: C0326: Exactly one space required after comma
    def test_append(self,first_value,second_value,third_value):
                                                 ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:359:32: C0326: Exactly one space required after comma
        [GateSequence([IGate()]),GateSequence([IGate(),IGate()]),0.0],
                                ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:359:54: C0326: Exactly one space required after comma
        [GateSequence([IGate()]),GateSequence([IGate(),IGate()]),0.0],
                                                      ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:359:64: C0326: Exactly one space required after comma
        [GateSequence([IGate()]),GateSequence([IGate(),IGate()]),0.0],
                                                                ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:360:30: C0326: Exactly one space required after comma
        [GateSequence([IGate(),IGate()]),GateSequence([IGate(),IGate(),IGate()]),0.0],
                              ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:360:40: C0326: Exactly one space required after comma
        [GateSequence([IGate(),IGate()]),GateSequence([IGate(),IGate(),IGate()]),0.0],
                                        ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:360:62: C0326: Exactly one space required after comma
        [GateSequence([IGate(),IGate()]),GateSequence([IGate(),IGate(),IGate()]),0.0],
                                                              ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:360:70: C0326: Exactly one space required after comma
        [GateSequence([IGate(),IGate()]),GateSequence([IGate(),IGate(),IGate()]),0.0],
                                                                      ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:360:80: C0326: Exactly one space required after comma
        [GateSequence([IGate(),IGate()]),GateSequence([IGate(),IGate(),IGate()]),0.0],
                                                                                ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:361:30: C0326: Exactly one space required after comma
        [GateSequence([IGate(),RXGate(1)]),GateSequence([RXGate(1)]),0.0],
                              ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:361:42: C0326: Exactly one space required after comma
        [GateSequence([IGate(),RXGate(1)]),GateSequence([RXGate(1)]),0.0],
                                          ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:361:68: C0326: Exactly one space required after comma
        [GateSequence([IGate(),RXGate(1)]),GateSequence([RXGate(1)]),0.0],
                                                                    ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:362:34: C0326: Exactly one space required after comma
        [GateSequence([RXGate(1)]),GateSequence([RXGate(0.99)]),0.01],
                                  ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:362:63: C0326: Exactly one space required after comma
        [GateSequence([RXGate(1)]),GateSequence([RXGate(0.99)]),0.01],
                                                               ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:363:0: C0330: Wrong hanging indentation.
          )
    |   | ^ (bad-continuation)
test/python/transpiler/test_solovay_kitaev.py:365:0: C0301: Line too long (126/100) (line-too-long)
test/python/transpiler/test_solovay_kitaev.py:365:43: C0326: Exactly one space required after comma
    def test_represents_same_gate_true(self,first_sequence: 'GateSequence',second_sequence: 'GateSequence', precision: float):
                                           ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:365:74: C0326: Exactly one space required after comma
    def test_represents_same_gate_true(self,first_sequence: 'GateSequence',second_sequence: 'GateSequence', precision: float):
                                                                          ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:367:0: C0303: Trailing whitespace (trailing-whitespace)
test/python/transpiler/test_solovay_kitaev.py:369:32: C0326: Exactly one space required after comma
        [GateSequence([IGate()]),GateSequence([IGate(),RXGate(1)]),0.0],
                                ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:369:54: C0326: Exactly one space required after comma
        [GateSequence([IGate()]),GateSequence([IGate(),RXGate(1)]),0.0],
                                                      ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:369:66: C0326: Exactly one space required after comma
        [GateSequence([IGate()]),GateSequence([IGate(),RXGate(1)]),0.0],
                                                                  ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:370:32: C0326: Exactly one space required after comma
        [GateSequence([RXGate(1),RXGate(1),RXGate(0.5)]),GateSequence([RXGate(1)]),0.0],
                                ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:370:42: C0326: Exactly one space required after comma
        [GateSequence([RXGate(1),RXGate(1),RXGate(0.5)]),GateSequence([RXGate(1)]),0.0],
                                          ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:370:56: C0326: Exactly one space required after comma
        [GateSequence([RXGate(1),RXGate(1),RXGate(0.5)]),GateSequence([RXGate(1)]),0.0],
                                                        ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:370:82: C0326: Exactly one space required after comma
        [GateSequence([RXGate(1),RXGate(1),RXGate(0.5)]),GateSequence([RXGate(1)]),0.0],
                                                                                  ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:371:34: C0326: Exactly one space required after comma
        [GateSequence([RXGate(1)]),GateSequence([RXGate(0.5)]),0.0],
                                  ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:371:62: C0326: Exactly one space required after comma
        [GateSequence([RXGate(1)]),GateSequence([RXGate(0.5)]),0.0],
                                                              ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:372:34: C0326: Exactly one space required after comma
        [GateSequence([RXGate(1)]),GateSequence([RXGate(0.3),RXGate(0.2)]),0.0],
                                  ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:372:60: C0326: Exactly one space required after comma
        [GateSequence([RXGate(1)]),GateSequence([RXGate(0.3),RXGate(0.2)]),0.0],
                                                            ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:372:74: C0326: Exactly one space required after comma
        [GateSequence([RXGate(1)]),GateSequence([RXGate(0.3),RXGate(0.2)]),0.0],
                                                                          ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:373:0: C0301: Line too long (104/100) (line-too-long)
test/python/transpiler/test_solovay_kitaev.py:373:34: C0326: Exactly one space required after comma
        [GateSequence([RXGate(0.3),RXGate(0.5),RXGate(1)]),GateSequence([RXGate(0.3),RXGate(0.2)]),0.0],
                                  ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:373:46: C0326: Exactly one space required after comma
        [GateSequence([RXGate(0.3),RXGate(0.5),RXGate(1)]),GateSequence([RXGate(0.3),RXGate(0.2)]),0.0],
                                              ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:373:58: C0326: Exactly one space required after comma
        [GateSequence([RXGate(0.3),RXGate(0.5),RXGate(1)]),GateSequence([RXGate(0.3),RXGate(0.2)]),0.0],
                                                          ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:373:84: C0326: Exactly one space required after comma
        [GateSequence([RXGate(0.3),RXGate(0.5),RXGate(1)]),GateSequence([RXGate(0.3),RXGate(0.2)]),0.0],
                                                                                    ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:373:98: C0326: Exactly one space required after comma
        [GateSequence([RXGate(0.3),RXGate(0.5),RXGate(1)]),GateSequence([RXGate(0.3),RXGate(0.2)]),0.0],
                                                                                                  ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:374:0: C0301: Line too long (104/100) (line-too-long)
test/python/transpiler/test_solovay_kitaev.py:374:34: C0326: Exactly one space required after comma
        [GateSequence([RXGate(0.3),RXGate(0.8),RXGate(1)]),GateSequence([RXGate(0.1),RXGate(0.2)]),0.0],
                                  ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:374:46: C0326: Exactly one space required after comma
        [GateSequence([RXGate(0.3),RXGate(0.8),RXGate(1)]),GateSequence([RXGate(0.1),RXGate(0.2)]),0.0],
                                              ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:374:58: C0326: Exactly one space required after comma
        [GateSequence([RXGate(0.3),RXGate(0.8),RXGate(1)]),GateSequence([RXGate(0.1),RXGate(0.2)]),0.0],
                                                          ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:374:84: C0326: Exactly one space required after comma
        [GateSequence([RXGate(0.3),RXGate(0.8),RXGate(1)]),GateSequence([RXGate(0.1),RXGate(0.2)]),0.0],
                                                                                    ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:374:98: C0326: Exactly one space required after comma
        [GateSequence([RXGate(0.3),RXGate(0.8),RXGate(1)]),GateSequence([RXGate(0.1),RXGate(0.2)]),0.0],
                                                                                                  ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:375:0: C0330: Wrong hanging indentation.
          )
    |   | ^ (bad-continuation)
test/python/transpiler/test_solovay_kitaev.py:377:0: C0301: Line too long (127/100) (line-too-long)
test/python/transpiler/test_solovay_kitaev.py:377:44: C0326: Exactly one space required after comma
    def test_represents_same_gate_false(self,first_sequence: 'GateSequence',second_sequence: 'GateSequence', precision: float):
                                            ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:377:75: C0326: Exactly one space required after comma
    def test_represents_same_gate_false(self,first_sequence: 'GateSequence',second_sequence: 'GateSequence', precision: float):
                                                                           ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:381:30: C0326: Exactly one space required after comma
        [GateSequence([IGate(),IGate()]),GateSequence([]),0.0],
                              ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:381:40: C0326: Exactly one space required after comma
        [GateSequence([IGate(),IGate()]),GateSequence([]),0.0],
                                        ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:381:57: C0326: Exactly one space required after comma
        [GateSequence([IGate(),IGate()]),GateSequence([]),0.0],
                                                         ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:382:30: C0326: Exactly one space required after comma
        [GateSequence([IGate(),RXGate(1),IGate()]),GateSequence([RXGate(1)]),0.0],
                              ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:382:40: C0326: Exactly one space required after comma
        [GateSequence([IGate(),RXGate(1),IGate()]),GateSequence([RXGate(1)]),0.0],
                                        ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:382:50: C0326: Exactly one space required after comma
        [GateSequence([IGate(),RXGate(1),IGate()]),GateSequence([RXGate(1)]),0.0],
                                                  ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:382:76: C0326: Exactly one space required after comma
        [GateSequence([IGate(),RXGate(1),IGate()]),GateSequence([RXGate(1)]),0.0],
                                                                            ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:383:0: C0301: Line too long (106/100) (line-too-long)
test/python/transpiler/test_solovay_kitaev.py:383:30: C0326: Exactly one space required after comma
        [GateSequence([IGate(),RXGate(1),IGate(),RXGate(0.4)]),GateSequence([RXGate(1),RXGate(0.4)]),0.0],
                              ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:383:40: C0326: Exactly one space required after comma
        [GateSequence([IGate(),RXGate(1),IGate(),RXGate(0.4)]),GateSequence([RXGate(1),RXGate(0.4)]),0.0],
                                        ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:383:48: C0326: Exactly one space required after comma
        [GateSequence([IGate(),RXGate(1),IGate(),RXGate(0.4)]),GateSequence([RXGate(1),RXGate(0.4)]),0.0],
                                                ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:383:62: C0326: Exactly one space required after comma
        [GateSequence([IGate(),RXGate(1),IGate(),RXGate(0.4)]),GateSequence([RXGate(1),RXGate(0.4)]),0.0],
                                                              ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:383:86: C0326: Exactly one space required after comma
        [GateSequence([IGate(),RXGate(1),IGate(),RXGate(0.4)]),GateSequence([RXGate(1),RXGate(0.4)]),0.0],
                                                                                      ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:383:100: C0326: Exactly one space required after comma
        [GateSequence([IGate(),RXGate(1),IGate(),RXGate(0.4)]),GateSequence([RXGate(1),RXGate(0.4)]),0.0],
                                                                                                    ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:384:30: C0326: Exactly one space required after comma
        [GateSequence([IGate(),RXGate(2*math.pi),RXGate(2*math.pi)]),GateSequence([]),1e10],
                              ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:384:48: C0326: Exactly one space required after comma
        [GateSequence([IGate(),RXGate(2*math.pi),RXGate(2*math.pi)]),GateSequence([]),1e10],
                                                ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:384:68: C0326: Exactly one space required after comma
        [GateSequence([IGate(),RXGate(2*math.pi),RXGate(2*math.pi)]),GateSequence([]),1e10],
                                                                    ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:384:85: C0326: Exactly one space required after comma
        [GateSequence([IGate(),RXGate(2*math.pi),RXGate(2*math.pi)]),GateSequence([]),1e10],
                                                                                     ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:387:0: C0301: Line too long (114/100) (line-too-long)
test/python/transpiler/test_solovay_kitaev.py:387:26: C0326: Exactly one space required after comma
    def test_simplify(self,original_sequence: 'GateSequence',expected_sequence: 'GateSequence', precision: float):
                          ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:387:60: C0326: Exactly one space required after comma
    def test_simplify(self,original_sequence: 'GateSequence',expected_sequence: 'GateSequence', precision: float):
                                                            ^ (bad-whitespace)
test/python/transpiler/test_solovay_kitaev.py:393:0: C0304: Final newline missing (missing-final-newline)
test/python/transpiler/test_solovay_kitaev.py:209:0: C0116: Missing function or method docstring (missing-function-docstring)
test/python/transpiler/test_solovay_kitaev.py:212:0: C0116: Missing function or method docstring (missing-function-docstring)
test/python/transpiler/test_solovay_kitaev.py:229:4: C0116: Missing function or method docstring (missing-function-docstring)
test/python/transpiler/test_solovay_kitaev.py:236:4: C0116: Missing function or method docstring (missing-function-docstring)
test/python/transpiler/test_solovay_kitaev.py:248:4: C0116: Missing function or method docstring (missing-function-docstring)
test/python/transpiler/test_solovay_kitaev.py:257:4: C0116: Missing function or method docstring (missing-function-docstring)
test/python/transpiler/test_solovay_kitaev.py:265:4: C0116: Missing function or method docstring (missing-function-docstring)
test/python/transpiler/test_solovay_kitaev.py:275:4: C0116: Missing function or method docstring (missing-function-docstring)
test/python/transpiler/test_solovay_kitaev.py:344:4: C0116: Missing function or method docstring (missing-function-docstring)
test/python/transpiler/test_solovay_kitaev.py:356:-1: W0105: String statement has no effect (pointless-string-statement)
test/python/transpiler/test_solovay_kitaev.py:364:4: C0116: Missing function or method docstring (missing-function-docstring)
test/python/transpiler/test_solovay_kitaev.py:376:4: C0116: Missing function or method docstring (missing-function-docstring)
test/python/transpiler/test_solovay_kitaev.py:386:4: C0116: Missing function or method docstring (missing-function-docstring)
test/python/transpiler/test_solovay_kitaev.py:24:0: W0611: Unused special_ortho_group imported from scipy.stats (unused-import)

------------------------------------------------------------------
Your code has been rated at 6.65/10 (previous run: 1.24/10, +5.41)
```
</details>

--- a/qiskit/transpiler/passes/synthesis/solovay_kitaev.py
+++ b/qiskit/transpiler/passes/synthesis/solovay_kitaev.py
@@ -55,7 +55,7 @@ class SolovayKitaev():
         products = []
         for reps in range(1, depth + 1):
             products += list(list(comb)
-                            for comb in itertools.product(*[basis_gates] * reps))
+                             for comb in itertools.product(*[basis_gates] * reps))
 
         sequences = []
         for item in products:
@@ -136,7 +136,6 @@ class SolovayKitaev():
         w_n1 = self._recurse(tuple_v_w[1], n - 1)
         return v_n1.dot(w_n1).dot(v_n1.adjoint()).dot(w_n1.adjoint()).dot(u_n1)
 
-
     def find_basic_approximation(self, sequence: GateSequence) -> Gate:
         """Finds gate in ``self._basic_approximations`` that best represents ``u``.
 
@@ -150,6 +149,7 @@ class SolovayKitaev():
             return np.linalg.norm(np.subtract(x.product, sequence.product))
 
         return min(self._basic_approximations, key=key)
+
 
 def commutator_decompose(u_so3: np.ndarray) -> Tuple[GateSequence, GateSequence]:
     """Decompose an SO(3)-matrix as a balanced commutator.
@@ -198,6 +198,7 @@ def commutator_decompose(u_so3: np.ndarray) -> Tuple[GateSequence, GateSequence]
     w = np.dot(np.dot(sim_matrix, wy), sim_matrix_dagger)
 
     return GateSequence.from_matrix(v), GateSequence.from_matrix(w)
+
 
 def _check_candidate(candidate: GateSequence, sequences: List[GateSequence]) -> bool:
     from qiskit.quantum_info.operators.predicates import matrix_equal

--- a/qiskit/transpiler/passes/synthesis/solovay_kitaev_utils.py
+++ b/qiskit/transpiler/passes/synthesis/solovay_kitaev_utils.py
@@ -387,45 +387,33 @@ def solve_decomposition_angle(matrix: np.ndarray) -> float:
     decomposition_angle = scipy.optimize.fsolve(objective, angle)[0]
     return decomposition_angle
 
-def compute_frobenius_norm(u: np.ndarray) -> float:
-    """Computes the Frobenius-norm of the matrix ``u``.
 
-    Args:
-        u: SO(3)-matrix.
-
-    Returns:
-        The Frobenius-norm of the matrix u.
-    """
-    return np.linalg.norm(u)
-
-
-def compute_euler_angles_from_s03(u: np.ndarray) -> Tuple[float, float, float]:
-    """Computes the Euler angles from the SO(3)-matrix u.
+def compute_euler_angles_from_s03(matrix: np.ndarray) -> Tuple[float, float, float]:
+    """Computes the Euler angles from the input SO(3)-matrix.
 
     Uses the algorithm from Gregory Slabaugh,
     see `here <https://www.gregslabaugh.net/publications/euler.pdf>`_.
 
     Args:
-        u: The SO(3)-matrix for which the Euler angles need to be computed.
+        matrix: The SO(3)-matrix for which the Euler angles need to be computed.
 
     Returns:
-        Tuple phi, theta, psi\n
-        where phi is rotation about z-axis, theta rotation about y-axis\n
-        and psi rotation about x-axis.
+        A tuple ``(phi, theta, psi)``, which indicate rotations about the Z, Y and X axis,
+        respectively.
     """
-    if (u[2][0] != 1 and u[2][1] != -1):
-        theta = -math.asin(u[2][0])
-        psi = math.atan2(u[2][1]/math.cos(theta), u[2][2]/math.cos(theta))
-        phi = math.atan2(u[1][0]/math.cos(theta), u[0][0]/math.cos(theta))
+    if matrix[2][0] != 1 and matrix[2][1] != -1:
+        theta = -math.asin(matrix[2][0])
+        psi = math.atan2(matrix[2][1]/math.cos(theta), matrix[2][2]/math.cos(theta))
+        phi = math.atan2(matrix[1][0]/math.cos(theta), matrix[0][0]/math.cos(theta))
         return phi, theta, psi
     else:
         phi = 0
-        if u[2][0] == 1:
+        if matrix[2][0] == 1:
             theta = math.pi/2
-            psi = phi + math.atan2(u[0][1], u[0][2])
+            psi = phi + math.atan2(matrix[0][1], matrix[0][2])
         else:
             theta = -math.pi/2
-            psi = -phi + math.atan2(-u[0][1], -u[0][2])
+            psi = -phi + math.atan2(-matrix[0][1], -matrix[0][2])
         return phi, theta, psi
 
 

--- a/test/python/transpiler/test_solovay_kitaev.py
+++ b/test/python/transpiler/test_solovay_kitaev.py
@@ -21,7 +21,6 @@ import scipy
 from hypothesis import given
 import hypothesis.strategies as st
 from scipy.optimize import minimize
-from scipy.stats import special_ortho_group
 from ddt import ddt, data, unpack
 import qiskit.circuit.library as gates
 
@@ -32,9 +31,10 @@ from qiskit.transpiler.passes import SolovayKitaevDecomposition
 from qiskit.transpiler.passes.synthesis import commutator_decompose
 from qiskit.test import QiskitTestCase
 from qiskit.quantum_info import Operator
-from qiskit.transpiler.passes.synthesis import GateSequence 
+from qiskit.transpiler.passes.synthesis import GateSequence
 
 # pylint: disable=invalid-name, missing-class-docstring
+
 
 class H(Gate):
     def __init__(self):
@@ -147,17 +147,28 @@ def distance(A, B):
     result2 = minimize(objective, [0.5], bounds=[(-np.pi, np.pi)])
     return min(result1.fun, result2.fun)
 
-def _generate_x_rotation(angle:float) -> np.ndarray:
-    return np.array([[1,0,0],[0,math.cos(angle),-math.sin(angle)],[0,math.sin(angle),math.cos(angle)]])
 
-def _generate_y_rotation(angle:float) -> np.ndarray:
-    return np.array([[math.cos(angle),0,math.sin(angle)],[0,1,0],[-math.sin(angle),0,math.cos(angle)]])
+def _generate_x_rotation(angle: float) -> np.ndarray:
+    return np.array([[1, 0, 0],
+                     [0, math.cos(angle), -math.sin(angle)],
+                     [0, math.sin(angle), math.cos(angle)]])
 
-def _generate_z_rotation(angle:float) -> np.ndarray:
-    return np.array([[math.cos(angle),-math.sin(angle),0],[math.sin(angle),math.cos(angle),0],[0,0,1]])
+
+def _generate_y_rotation(angle: float) -> np.ndarray:
+    return np.array([[math.cos(angle), 0, math.sin(angle)],
+                     [0, 1, 0],
+                     [-math.sin(angle), 0, math.cos(angle)]])
+
+
+def _generate_z_rotation(angle: float) -> np.ndarray:
+    return np.array([[math.cos(angle), -math.sin(angle), 0],
+                     [math.sin(angle), math.cos(angle), 0],
+                     [0, 0, 1]])
+
 
 def _generate_random_rotation() -> np.ndarray:
     return np.array(scipy.stats.special_ortho_group.rvs(3))
+
 
 def _build_rotation(angle: float, axis: int) -> np.ndarray:
     if axis == 0:
@@ -169,51 +180,71 @@ def _build_rotation(angle: float, axis: int) -> np.ndarray:
     else:
         return _generate_random_rotation()
 
+
 def _build_axis(axis: int) -> np.ndarray:
     if axis == 0:
-        return np.array([1.0,0.0,0.0])
+        return np.array([1.0, 0.0, 0.0])
     elif axis == 1:
-        return np.array([0.0,1.0,0.0])
+        return np.array([0.0, 1.0, 0.0])
     elif axis == 2:
-        return np.array([0.0,0.0,1.0])
+        return np.array([0.0, 0.0, 1.0])
     else:
-        return np.array([1/math.sqrt(3),1/math.sqrt(3),1/math.sqrt(3)])
+        return np.array([1/math.sqrt(3), 1/math.sqrt(3), 1/math.sqrt(3)])
 
-def _generate_x_su2(angle:float) -> np.ndarray:
+
+def _generate_x_su2(angle: float) -> np.ndarray:
     return np.array([[math.cos(angle/2), math.sin(angle/2)*1j],
-                       [math.sin(angle/2)*1j, math.cos(angle/2)]], dtype=complex)
+                     [math.sin(angle/2)*1j, math.cos(angle/2)]], dtype=complex)
 
-def _generate_y_su2(angle:float) -> np.ndarray:
+
+def _generate_y_su2(angle: float) -> np.ndarray:
     return np.array([[math.cos(angle/2), math.sin(angle/2)],
-                         [-math.sin(angle/2), math.cos(angle/2)]], dtype=complex)
+                     [-math.sin(angle/2), math.cos(angle/2)]], dtype=complex)
 
-def _generate_z_su2(angle:float) -> np.ndarray:
+
+def _generate_z_su2(angle: float) -> np.ndarray:
     return np.array([[np.exp(-(1/2)*angle*1j), 0], [0, np.exp((1/2)*angle*1j)]], dtype=complex)
 
+
 def _generate_su2(alpha: complex, beta: complex) -> np.ndarray:
-    base = np.array([[alpha,beta],[-np.conj(beta),np.conj(alpha)]])
+    base = np.array([[alpha, beta], [-np.conj(beta), np.conj(alpha)]])
     det = np.linalg.det(base)
-    if abs(det)<1e10:
-        return np.array([[1,0],[0,1]])
+    if abs(det) < 1e10:
+        return np.array([[1, 0], [0, 1]])
     else:
         return np.linalg.det(base)*base
 
+
 def _build_unit_vector(a: float, b: float, c: float) -> np.ndarray:
-    vector = np.array([a,b,c])
-    if a != 0.0 or b != 0.0 or c!= 0.0:
+    vector = np.array([a, b, c])
+    if a != 0.0 or b != 0.0 or c != 0.0:
         unit_vector = vector/np.linalg.norm(vector)
         return unit_vector
     else:
-        return np.array([1,0,0])
+        return np.array([1, 0, 0])
+
 
 def is_so3_matrix(array: np.ndarray) -> bool:
-    return array.shape == (3,3) and abs(np.linalg.det(array)-1.0)< 1e-10 and not False in np.isreal(array) 
+    """Check if the input array is a SO(3) matrix."""
+    if array.shape != (3, 3):
+        return False
+
+    if abs(np.linalg.det(array)-1.0) < 1e-10:
+        return False
+
+    if False in np.isreal(array):
+        return False
+
+    return True
+
 
 def are_almost_equal_so3_matrices(a: np.ndarray, b: np.ndarray) -> bool:
-    for t in itertools.product(range(2),range(2)):
-        if abs(a[t[0]][t[1]]-b[t[0]][t[1]])> 1e-10:
+    """TODO"""
+    for t in itertools.product(range(2), range(2)):
+        if abs(a[t[0]][t[1]]-b[t[0]][t[1]]) > 1e-10:
             return False
     return True
+
 
 @ddt
 class TestSolovayKitaev(QiskitTestCase):
@@ -223,17 +254,20 @@ class TestSolovayKitaev(QiskitTestCase):
         [_generate_x_rotation(0.1)],
         [_generate_y_rotation(0.2)],
         [_generate_z_rotation(0.3)],
-        [np.dot(_generate_z_rotation(0.5),_generate_y_rotation(0.4))],
-        [np.dot(_generate_y_rotation(0.5),_generate_x_rotation(0.4))]
+        [np.dot(_generate_z_rotation(0.5), _generate_y_rotation(0.4))],
+        [np.dot(_generate_y_rotation(0.5), _generate_x_rotation(0.4))]
     )
     @unpack
-    def test_commutator_decompose_returns_tuple_of_two_so3_gatesequences(self, u_so3: np.ndarray):        
+    def test_commutator_decompose_returns_tuple_of_two_so3_gatesequences(self, u_so3: np.ndarray):
+        """Test that ``commutator_decompose`` returns two SO(3) gate sequences."""
         actual_result = commutator_decompose(u_so3)
         self.assertTrue(is_so3_matrix(actual_result[0].product))
         self.assertTrue(is_so3_matrix(actual_result[1].product))
 
-    @given(st.builds(_build_rotation,st.floats(max_value=2*math.pi,min_value=0),st.integers(min_value=0,max_value=4)))
-    def test_commutator_decompose_returns_tuple_of_two_so3_gatesequences_2(self, u_so3: np.ndarray):        
+    @given(st.builds(_build_rotation, st.floats(max_value=2*math.pi, min_value=0),
+                     st.integers(min_value=0, max_value=4)))
+    def test_commutator_decompose_returns_tuple_of_two_so3_gatesequences_2(self, u_so3: np.ndarray):
+        """Test that ``commutator_decompose`` returns two SO(3) gate sequences."""
         actual_result = commutator_decompose(u_so3)
         self.assertTrue(is_so3_matrix(actual_result[0].product))
         self.assertTrue(is_so3_matrix(actual_result[1].product))
@@ -242,45 +276,53 @@ class TestSolovayKitaev(QiskitTestCase):
         [_generate_x_rotation(0.1)],
         [_generate_y_rotation(0.2)],
         [_generate_z_rotation(0.3)],
-        [np.dot(_generate_z_rotation(0.5),_generate_y_rotation(0.4))],
-        [np.dot(_generate_y_rotation(0.5),_generate_x_rotation(0.4))]
+        [np.dot(_generate_z_rotation(0.5), _generate_y_rotation(0.4))],
+        [np.dot(_generate_y_rotation(0.5), _generate_x_rotation(0.4))]
     )
     @unpack
-    def test_commutator_decompose_returns_tuple_whose_commutator_equals_input(self, u_so3: np.ndarray):        
+    def test_commutator_decompose_returns_tuple_whose_commutator_equals_input(self, u_so3):
+        """Test that ``commutator_decompose`` exactly decomposes the input."""
         actual_result = commutator_decompose(u_so3)
         first_so3 = actual_result[0].product
         second_so3 = actual_result[1].product
-        actual_commutator = np.dot(first_so3,np.dot(second_so3,np.dot(np.matrix.getH(first_so3),np.matrix.getH(second_so3))))
-        self.assertTrue(are_almost_equal_so3_matrices( actual_commutator,u_so3))
+        actual_commutator = np.dot(first_so3, np.dot(second_so3, np.dot(
+            np.matrix.getH(first_so3), np.matrix.getH(second_so3))))
+        self.assertTrue(are_almost_equal_so3_matrices(actual_commutator, u_so3))
 
-    @given(st.builds(_build_rotation,st.floats(max_value=2*math.pi,min_value=0),st.integers(min_value=0,max_value=4)))
-    def test_commutator_decompose_returns_tuple_whose_commutator_equals_input_2(self, u_so3: np.ndarray):        
+    @given(st.builds(_build_rotation, st.floats(max_value=2*math.pi, min_value=0),
+                     st.integers(min_value=0, max_value=4)))
+    def test_commutator_decompose_returns_tuple_whose_commutator_equals_input_2(self, u_so3):
+        """Test that ``commutator_decompose`` exactly decomposes the input."""
         actual_result = commutator_decompose(u_so3)
         first_so3 = actual_result[0].product
         second_so3 = actual_result[1].product
-        actual_commutator = np.dot(first_so3,np.dot(second_so3,np.dot(np.matrix.getH(first_so3),np.matrix.getH(second_so3))))
-        self.assertTrue(are_almost_equal_so3_matrices( actual_commutator,u_so3))
+        actual_commutator = np.dot(first_so3, np.dot(second_so3, np.dot(
+            np.matrix.getH(first_so3), np.matrix.getH(second_so3))))
+        self.assertTrue(are_almost_equal_so3_matrices(actual_commutator, u_so3))
 
-    @given(st.builds(_build_rotation,st.floats(max_value=2*math.pi,min_value=0),st.integers(min_value=0,max_value=4)))
-    def test_commutator_decompose_returns_tuple_with_first_x_axis_rotation(self, u_so3: np.ndarray):
+    @given(st.builds(_build_rotation, st.floats(max_value=2*math.pi, min_value=0),
+                     st.integers(min_value=0, max_value=4)))
+    def test_commutator_decompose_returns_tuple_with_first_x_axis_rotation(self, u_so3):
+        """Test that ``commutator_decompose`` returns a X-rotation as first element."""
         actual_result = commutator_decompose(u_so3)
         actual = actual_result[0]
-        self.assertAlmostEqual(actual[0][0],1.0)
-        self.assertAlmostEqual(actual[0][1],0.0)
-        self.assertAlmostEqual(actual[0][2],0.0)
-        self.assertAlmostEqual(actual[1][0],0.0)
-        self.assertAlmostEqual(actual[2][0],0.0)
+        self.assertAlmostEqual(actual[0][0], 1.0)
+        self.assertAlmostEqual(actual[0][1], 0.0)
+        self.assertAlmostEqual(actual[0][2], 0.0)
+        self.assertAlmostEqual(actual[1][0], 0.0)
+        self.assertAlmostEqual(actual[2][0], 0.0)
 
-    @given(st.builds(_build_rotation,st.floats(max_value=2*math.pi,min_value=0),st.integers(min_value=0,max_value=4)))
-    def test_commutator_decompose_returns_tuple_with_second_y_axis_rotation(self, u_so3: np.ndarray):
+    @given(st.builds(_build_rotation, st.floats(max_value=2*math.pi, min_value=0),
+                     st.integers(min_value=0, max_value=4)))
+    def test_commutator_decompose_returns_tuple_with_second_y_axis_rotation(self, u_so3):
+        """Test that ``commutator_decompose`` returns a Y-rotation as second element."""
         actual_result = commutator_decompose(u_so3)
         actual = actual_result[1]
-        self.assertAlmostEqual(actual[1][1],1.0)
-        self.assertAlmostEqual(actual[0][1],0.0)
-        self.assertAlmostEqual(actual[1][0],0.0)
-        self.assertAlmostEqual(actual[1][2],0.0)
-        self.assertAlmostEqual(actual[2][1],0.0)
-    
+        self.assertAlmostEqual(actual[1][1], 1.0)
+        self.assertAlmostEqual(actual[0][1], 0.0)
+        self.assertAlmostEqual(actual[1][0], 0.0)
+        self.assertAlmostEqual(actual[1][2], 0.0)
+        self.assertAlmostEqual(actual[2][1], 0.0)
 
     def test_example(self):
         """@Lisa Example to show how to call the pass."""
@@ -301,6 +343,7 @@ class TestSolovayKitaev(QiskitTestCase):
         """@Lisa Example to show how to call the pass."""
         circuit = QuantumCircuit(1)
         circuit.rx(0.8, 0)
+        print(circuit.draw())
 
         basic_gates = [H(), T(), S(), T_dg(), S_dg()]
         synth = SolovayKitaevDecomposition(2, basic_gates)
@@ -339,52 +382,49 @@ class TestSolovayKitaev(QiskitTestCase):
 class TestSolovayKitaevUtils(QiskitTestCase):
     """Test the algebra utils."""
 
-    @data([GateSequence([IGate()]),IGate(),GateSequence([IGate(),IGate()])])
+    @data([GateSequence([IGate()]), IGate(), GateSequence([IGate(), IGate()])])
     @unpack
-    def test_append(self,first_value,second_value,third_value):
+    def test_append(self, first_value, second_value, third_value):
+        """Test append of ``GateSequence``."""
         actual_gate = first_value.append(second_value)
         self.assertTrue(actual_gate == third_value)
 
-    """
-    @data([GateSequence([IGate()]),GateSequence([TGate()]),GateSequence([IGate(),TGate()])],
-          [GateSequence([IGate()]),GateSequence([TGate(),IGate()]),GateSequence([IGate(),TGate(),IGate()])],
-          [GateSequence([IGate(),TGate(),RXGate(0.1)]),GateSequence([TGate(),IGate()]),GateSequence([IGate(),TGate(),RXGate(0.1),TGate(),IGate()])])
+    @data(
+        [GateSequence([IGate()]), GateSequence([IGate(), IGate()]), 0.0],
+        [GateSequence([IGate(), IGate()]), GateSequence([IGate(), IGate(), IGate()]), 0.0],
+        [GateSequence([IGate(), RXGate(1)]), GateSequence([RXGate(1)]), 0.0],
+        [GateSequence([RXGate(1)]), GateSequence([RXGate(0.99)]), 0.01],
+    )
     @unpack
-    def test_append(self,first_value,second_value,third_value):
-        actual_gate = first_value + second_value
-        self.assertTrue(actual_gate == third_value)
-    """
+    def test_represents_same_gate_true(self, first_sequence, second_sequence, precision):
+        """Test ``represents_same_gate``."""
+        self.assertTrue(first_sequence.represents_same_gate(second_sequence, precision))
 
     @data(
-        [GateSequence([IGate()]),GateSequence([IGate(),IGate()]),0.0],
-        [GateSequence([IGate(),IGate()]),GateSequence([IGate(),IGate(),IGate()]),0.0],
-        [GateSequence([IGate(),RXGate(1)]),GateSequence([RXGate(1)]),0.0],
-        [GateSequence([RXGate(1)]),GateSequence([RXGate(0.99)]),0.01],
-          )
+        [GateSequence([IGate()]), GateSequence([IGate(), RXGate(1)]), 0.0],
+        [GateSequence([RXGate(1), RXGate(1), RXGate(0.5)]), GateSequence([RXGate(1)]), 0.0],
+        [GateSequence([RXGate(1)]), GateSequence([RXGate(0.5)]), 0.0],
+        [GateSequence([RXGate(1)]), GateSequence([RXGate(0.3), RXGate(0.2)]), 0.0],
+        [GateSequence([RXGate(0.3), RXGate(0.5), RXGate(1)]),
+         GateSequence([RXGate(0.3), RXGate(0.2)]), 0.0],
+        [GateSequence([RXGate(0.3), RXGate(0.8), RXGate(1)]),
+         GateSequence([RXGate(0.1), RXGate(0.2)]), 0.0],
+    )
     @unpack
-    def test_represents_same_gate_true(self,first_sequence: 'GateSequence',second_sequence: 'GateSequence', precision: float):
-        self.assertTrue(first_sequence.represents_same_gate(second_sequence, precision))
-    
-    @data(
-        [GateSequence([IGate()]),GateSequence([IGate(),RXGate(1)]),0.0],
-        [GateSequence([RXGate(1),RXGate(1),RXGate(0.5)]),GateSequence([RXGate(1)]),0.0],
-        [GateSequence([RXGate(1)]),GateSequence([RXGate(0.5)]),0.0],
-        [GateSequence([RXGate(1)]),GateSequence([RXGate(0.3),RXGate(0.2)]),0.0],
-        [GateSequence([RXGate(0.3),RXGate(0.5),RXGate(1)]),GateSequence([RXGate(0.3),RXGate(0.2)]),0.0],
-        [GateSequence([RXGate(0.3),RXGate(0.8),RXGate(1)]),GateSequence([RXGate(0.1),RXGate(0.2)]),0.0],
-          )
-    @unpack
-    def test_represents_same_gate_false(self,first_sequence: 'GateSequence',second_sequence: 'GateSequence', precision: float):
+    def test_represents_same_gate_false(self, first_sequence, second_sequence, precision):
+        """Test ``represents_same_gate``."""
         self.assertFalse(first_sequence.represents_same_gate(second_sequence, precision))
 
     @data(
-        [GateSequence([IGate(),IGate()]),GateSequence([]),0.0],
-        [GateSequence([IGate(),RXGate(1),IGate()]),GateSequence([RXGate(1)]),0.0],
-        [GateSequence([IGate(),RXGate(1),IGate(),RXGate(0.4)]),GateSequence([RXGate(1),RXGate(0.4)]),0.0],
-        [GateSequence([IGate(),RXGate(2*math.pi),RXGate(2*math.pi)]),GateSequence([]),1e10],
+        [GateSequence([IGate(), IGate()]), GateSequence([]), 0.0],
+        [GateSequence([IGate(), RXGate(1), IGate()]), GateSequence([RXGate(1)]), 0.0],
+        [GateSequence([IGate(), RXGate(1), IGate(), RXGate(0.4)]),
+         GateSequence([RXGate(1), RXGate(0.4)]), 0.0],
+        [GateSequence([IGate(), RXGate(2*math.pi), RXGate(2*math.pi)]), GateSequence([]), 1e10],
     )
     @unpack
-    def test_simplify(self,original_sequence: 'GateSequence',expected_sequence: 'GateSequence', precision: float):
+    def test_simplify(self, original_sequence, expected_sequence, precision):
+        """Test the simplify method on ``GateSequence``."""
         actual_sequence = original_sequence.simplify(precision)
         self.assertTrue(actual_sequence == expected_sequence)
 

